src/lru_cache.rs

```diff
@@ -100,16 +63,10 @@ impl<K: Hash + Eq, V> LruCache<K, V> {
     /// ```
     #[unstable = "matches collection reform specification, waiting for dust to settle"]
     pub fn new(capacity: uint) -> LruCache<K, V> {
-        let cache = LruCache {
-            map: HashMap::new(),
+        LruCache {
+            map: LinkedHashMap::new(),
```

Comment:
I think, it is safer to use default capacity, because LruCache might not be ever used at full capacity, not need to waste memory.